plugins/citadel-paper/src/main/java/vg/civcraft/mc/citadel/listener/BlockListener.java

```diff
@@ -361,40 +363,51 @@ public void removeReinforcedAir(BlockPlaceEvent e) {
         }
     }
 
-    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
-    public void preventStrippingLogs(PlayerInteractEvent pie) {
-        if (!pie.hasBlock()) {
-            return;
-        }
-        if (pie.getAction() != Action.RIGHT_CLICK_BLOCK) {
-            return;
-        }
-        Block block = pie.getClickedBlock();
-        if (!MoreTags.LOGS.isTagged(block.getType())) {
+    @EventHandler(
+        priority = EventPriority.LOW,
+        ignoreCancelled = true
+    )
+    private void preventStrippingLogs(
+        final @NotNull PlayerInteractEvent event
+    ) {
+        if (event.getAction() != Action.RIGHT_CLICK_BLOCK) {
             return;
         }
-        EquipmentSlot hand = pie.getHand();
-        if (hand != EquipmentSlot.HAND && hand != EquipmentSlot.OFF_HAND) {
+        final Block block = event.getClickedBlock();
+        if (block == null) { // This shouldn't happen, but just in case
             return;
         }
-        ItemStack relevant;
-        Player p = pie.getPlayer();
-        if (hand == EquipmentSlot.HAND) {
-            relevant = p.getInventory().getItemInMainHand();
-        } else {
-            relevant = p.getInventory().getItemInOffHand();
+        switch (block.getType()) {
+            case ACACIA_LOG:
+            case BIRCH_LOG:
+            case CHERRY_LOG:
+            case DARK_OAK_LOG:
+            case JUNGLE_LOG:
+            case MANGROVE_LOG:
+            case OAK_LOG:
+            case SPRUCE_LOG:
+                break;
+            default:
+                return;
```

Comment:
whats the point in checking the block type here? also doesn't this add another point where a blocklist needs to be maintained?

Reply:
It's necessary to check the block type here because there is no `LogStripEvent` or equivalent, so the event needs to be inferred from a right-click block interaction involving a log and an axe. And while this switch will need to be maintained, the event handler is no longer vulnerable to collateral changes.